server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala

```diff
@@ -183,8 +183,12 @@ class ClientListenerRegistry(configuration: HotRodServerConfiguration) extends L
       @CacheEntryExpired
       def onCacheEvent(event: CacheEntryEvent[Bytes, Bytes]) {
          if (isSendEvent(event)) {
-            val dataVersion = event.getMetadata.version().asInstanceOf[NumericVersion].getVersion
-            sendEvent(event.getKey, event.getValue, dataVersion, event)
+            sendEvent(event.getKey, event.getValue, if (event.getMetadata == null) {
```

Comment:
Cool, I will update it.   The issue is not with having versioning enabled or not.  The problem is that there are some cases where the metadata and value are not available from an expiration event and thus we have to handle those null values.